src/components/publishing/share.tsx

```diff
@@ -1,13 +1,41 @@
 import React from "react"
-import track from "react-tracking"
 import styled from "styled-components"
 import Icon from "../icon"
 
+import { track as _track, TrackingInfo, trackWithoutProps, trackWithProps } from "../../utils/track"
+
 interface ShareProps extends React.HTMLProps<HTMLDivElement> {
   url: string
   title: string
 }
 
+/**
+ * An example of a typed `track` function that does not have types for the props used by this component and only uses
+ * the default global tracking-info types.
+ *
+ * @note This weird `import { track as _track } from "../../utils/track"` dance is only done to have this comment in the
+ *       same format as the ones below.
+ */
+const track = _track
+
+/**
+ * An example of a typed `track` function that does not have types for the props used by this component, but does allow
+ * types for the tracking-info.
+ */
+// const track = trackWithoutProps<TrackingInfo.Shareable>()
+
+/**
+ * An example of a typed `track` function that has types for the props used by this component, but only uses the default
+ * global tracking-info types.
+ */
+// const track = trackWithProps<ShareProps>()
+
+/**
+ * An example of a typed `track` function that has types for the props used by this component and specifies extended
+ * types for tracking-info, in this case the centralized types shareable tracking-info.
+ */
+// const track = trackWithProps<ShareProps, TrackingInfo.Shareable>()
```

Comment:
These are all just examples of the various possible permutations with out own `track` module, which augments the generic `react-tracking` types.

Reply:
@craigspaeth So when you add actual types for all publishing components, please remove these 🙏